client/signup/controller.js

```diff
@@ -112,14 +113,17 @@ export default {
 		next();
 	},
 
-	start( context, next ) {
-		const basePath = sectionify( context.path ),
-			flowName = getFlowName( context.params ),
-			stepName = getStepName( context.params ),
-			stepSectionName = getStepSectionName( context.params );
+	async start( context, next ) {
+		const basePath = sectionify( context.path );
+		const flowName = getFlowName( context.params );
+		const stepName = getStepName( context.params );
+		const stepSectionName = getStepSectionName( context.params );
 
 		const { query } = initialContext;
 
+		// wait for the step component module to load
+		const stepComponent = await getStepComponent( stepName );
```

Comment:
I would have to look more closely to the signup controller as it's a fairly new piece to me, but I would imagine we could do something here in start that would essentially do something like:

```js
async start( context, next ) {

dispatchSomeIsLoadingAction();

const stepComponent = await getStepComponent( stepName );

dispatchSomeIsDoneLoadingAction();

```

Then we could factor that in to the steps themselves to show some indication/busy state.

Reply:
Just tested on slow 3g and can confirm that some loading indicator _might_ be nice as I was stuck on a step for a few seconds after submitting the form.

Reply:
I'll see what I can do about the loading indicator 🙂

Reply:
> or fall off my donkey mid-flow

Make sure you're wearing a helmet please @ramonjd!

> I'll see what I can do about the loading indicator

I'm happy to help with that if I can :)

Reply:
@spen @ramonjd What do you think about merging this PR and adressing the loading indicator issue in a follow-up one? I think this PR is already a major improvement -- do we need to wait until it becomes perfect?

Reply:
I think that makes sense too - I just wanted to see what you preferred :)

Reply:
(Consider my approval still effective, btw)